### ISSUE TITLE
Feature: Add sidebar to article view

### ISSUE THREAD

**Issue by mcwumbly** [erlend-sh/learndiscourse#1]
Navigation for easy access to articles from beginning to end.

Basic sidebar will display the same list of articles that the user first sees when they click on the portal index page while viewing an article within the same section.

**Comment by mcwumbly** [comment]
basics are done via b61a3f31911a340478bc3ffecbda01531ced6d0b

But some more work is needed still on styling...

**Comment by erlend-sh** [comment]
I hope we'll be able to fix the order from
- Administrators
- Designers
- Developers
- Moderators
- Sysadmins
- Users

to 
- Users
- Moderators
- Administrators
- Designers
- Developers
- Sysadmins

So it matches the sections order on `index.html`

For v1, how about if we only show the sidebar in pages? Not on `index.html`, not on the section pages.

**Comment by mcwumbly** [comment]
made it so sidebar only shows up on article pages.  will look into the ordering thing later...

**Comment by erickguan** [comment]
1. It's easy to run out of space if shows up article titles. I am not a big fan of showing up all the articles names in the sidebar instead of only sections and sub sections.
2. It seems we can add number prefix to order them?

**Comment by mcwumbly** [comment]
It seems a lot less useful if you don't show individual articles.

We can be more rigorous in our subsectioning to help.

And I'm thinking of allowing the sidebar to scroll independently.

**Comment by erickguan** [comment]
I know what you mean. But I don't like a scrollable sidebar instead of an affix effect(like bootstrap site). It would allow an always on navigation system.

BTW, why disable the sidebar for section?

And it seems I don't need to take care of styling problems after finished it.

**Comment by erlend-sh** [comment]
Mockups! I guess I basically ripped off [docs.transifex.com](http://docs.transifex.com/introduction/) (except for the sticky footer). Previously I suggested we keep the sidebar exclusively on pages to simplify things, but it probably confuses more than it simplifies. Here's how I see the sidebar fitting in sitewide:

### Index

![sidebar-portal](https://cloud.githubusercontent.com/assets/583842/8485565/be88d564-20fe-11e5-9997-a71e1b9c190a.jpg)
(Note that I moved and restyled the language selector a tiny bit, because currently it comes off as a heading)

### Section alt-1

![sidebar-section](https://cloud.githubusercontent.com/assets/583842/8485543/99fd18cc-20fe-11e5-98c9-36792cdaac00.jpg)

### Section alt-2

![sidebar-section-alt](https://cloud.githubusercontent.com/assets/583842/8485548/9f179238-20fe-11e5-97bb-38ff17605310.jpg)

### Page

![sidebar-page](https://cloud.githubusercontent.com/assets/583842/8485558/b8b27c30-20fe-11e5-98f3-b26f232eefdb.jpg)
- Made the text smaller and removed link styling
- Added simple grey background to set sidebar more apart.
- Suggested two different alternatives for sidebar-on-section. I figure just go with whatever is easiest.
- I would also like a independently scrolling sidebar. That seems like the simplest, 'most likely to work everywhere'-solution for v1. I definitely don't want to compromise by creating more subsections.

**Comment by ghost** [comment]
I personally prefer the old navbar we had that looked like the one from discourse.org. Why did we change it?

**Comment by erlend-sh** [comment]
It's not changed yet. Just a suggestion.

When you say "like the one from discourse.org", what are you referring to? Could you make a screenshot?

**Comment by ghost** [comment]
Its already changed on http://learndiscourse.org/

![http://i.imgur.com/UPQ1hV5.png](http://i.imgur.com/UPQ1hV5.png)

**Comment by erlend-sh** [comment]
Didn't understand what you were saying until I revisited this now. You're talking about the _top_ navbar. Aha! Well, we might go back to mimicking the official Discourse.org top-navbar eventually, but as long as we're unofficial, we shouldn't pretend like we're part of the main organisation.

Also that's a separate issue ;) This discussion is about the sidebar. The top-navbar just happened to make an appearance in the mockups.

**Comment by erlend-sh** [comment]
@mcwumbly did you agree/disagree with my other styling suggestions?

i.e.:
- Narrower menu width (we'll compensate for this with smaller text font and (maybe not until v2) shortened titles.
- Smaller text
- No link (blue) styling.

Also, I'm still in favour of showing the Sidebar on the index as well. I feel it ties the site together nicely, since it kind of doubles as a breadcrumbs, and for people who already know where they want to go, it's quicker than potentially having to scroll down in order to click on Developers for instance.

**Comment by erlend-sh** [comment]
(Oh, I thought I could assign multiple people :crying_cat_face: )

It appears the sidebar has gone through a lot of changes. Could someone run me through the basics of how it works now?
- Does it require manual action when new pages are added or is it completely automated?
- How does it deal with ordering?

**Comment by erickguan** [comment]
> Does it require manual action when new pages are added or is it completely automated?

You need to run two scripts as needed. And copy & paste files.

> How does it deal with ordering?

`weight` in every doc meta info is used to order things. So you can put a weight value on them. More weight will show higher.

**Comment by erlend-sh** [comment]
> You need to run two scripts as needed. And copy & paste files.

Figured it out, works fine!

> weight in every doc meta info is used to order things

I don't know where to look for this. Is there an example of it in use somewhere?

I think only two minor things remain with the sidebar now:
1. Sub-sections should be indented a bit, similar to what we had before, and the mockups above.
2. Currently active section/sub-section should be bold.

**Comment by erickguan** [comment]
Assuming we have three files on the `_en/administrators/essentials`: a, b, c.

a:

```
---
title: a
---
```

b:

```
---
title: b
weight: 3
---
```

c:

```
---
title: c
weight: 9999
----
```

Three docs should order like this: c, b, a. Since the c has the most highest weight, then is b. a has the default weight which is 0.

As for the sidebar styles, sure. will do that.

**Comment by erlend-sh** [comment]
Oh I see. So we're talking about the ordering of _articles_ in _portal pages_. That confused me a bit, since articles aren't currently present in the sidebar, hence not technically part of this issue :stuck_out_tongue: I'll open another topic for that, since it ties in with the new import script.

Back on topic about the sidebar, do we have a way of ordering the **sub-sections**? Say for example in Developers, I want to order the sub-sections environment, forum-import and hacking as:
1. Hacking
2. Forum Import
3. Environment

is there a way to do that or are we bound to alphabetical ordering when it comes to the sidebar ordering?

**Comment by mcwumbly** [comment]
It'd be great if the ordering of sections, subsection and articles that you implicitly specify in the `doc_list.yml` file were just reflected as in in the side bar.  How hard would that be to do @fantasticfears ?

**Comment by erickguan** [comment]
I think it's the behaviour now? All you need to do is order the section and subsection in the `doc_list.yml` file. However, you need to put weight in the meta attr for ordering doc.

**Comment by erlend-sh** [comment]
> I think it's the behaviour now? All you need to do is order the section and subsection in the doc_list.yml file.

Yup, that works! Sorry for the confusion, I should've tested that out before asking, just had to jump on a :train: 

> However, you need to put weight in the meta attr for ordering doc.

Got it. We can continue discussing that over in #20 .
